crates/neon/src/lib.rs

```diff
@@ -98,3 +98,10 @@ pub use neon_macros::*;
 
 #[cfg(feature = "napi-6")]
 mod lifecycle;
+
+#[cfg(feature = "napi-8")]
+static MODULE_TAG: once_cell::sync::Lazy<crate::sys::TypeTag> = once_cell::sync::Lazy::new(|| {
+    let (lower, upper) = rand::random::<(u64, u64)>();
```

Comment:
Is there another way to get a unique identifier for a shared library? We don't really need this to be dynamic, we just need it to be unique to this shared library.

It could be generated at build time, but a build script seemed a little heavy.

Reply:
Wouldn't a random, static identifier interfere with reproducible builds?

Reply:
That's a good reason to generate at runtime instead of build time.

Do you know if there's some cargo variable that could be used here? What we're actually asserting is the ABI stability of std::any::Any.

Reply:
I believe that it could still be generated at buildtime, FWIW. What if it was implemented like this?

```rust
use std::{path::PathBuf, process::Command};

fn env_var(key: &'static str) -> String {
    std::env::var(key).unwrap()
}

fn main() {
    let out_path = PathBuf::from(env_var("OUT_DIR"));
    let out_file = out_path.join("type_tag.rs");

    let rustc_version = Command::new(env_var("RUSTC")).arg("--version").map(|out| String::from_utf8(out.stdout).unwrap()).expect("failed to run rustc");
    
    let type_tag = format!(
        "crate::sys::TypeTag {{ rustc: {} }}",
        rustc_version
    );

    fs::write(out_file, type_tag).unwrap();

}
```

This should be enough to guarantee that an `Any` definition will be stable across all crates. I don't believe the toolchain is needed because Node modules can only be loaded on a single architecture, so the OS loader prevents any issues there 🤔. If I'm wrong, the contents of `TARGET` could be included as well.

Reply:
There are multiple toolchains that can target the same system. For example, msvc and mingw.

This is a pretty good idea, but it would need to be hashed. Node-API type tags are a u128.

Reply:
What if we hashed the rustc version into one `u64` and then the toolchain/target information into another using [DefaultHasher](https://doc.rust-lang.org/std/collections/hash_map/struct.DefaultHasher.html)?

Reply:
Ideally we wouldn't use the full 128-bits. I would like to reserve at least a byte for some static version identifier.

Hashing Rust metadata feels a little brittle to me. Do you happen to know if there's something that can be used for a unique identifier of a shared library that's been open with `dlopen`? Memory address or something?